server/async_tcp.go

```diff
@@ -142,9 +183,20 @@ func (s *AsyncServer) Run(ctx context.Context) error {
 		return err
 	}
 
-	err = s.eventLoop(ctx)
+	eventLoopCtx, cancelEventLoop := context.WithCancel(ctx)
+	defer cancelEventLoop()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err = s.eventLoop(eventLoopCtx)
+		if err != nil {
+			cancelEventLoop()
```

Comment:
Wouldn't the be done as part of defer. We may instead invoke s.InitiateShutdown() here.

Reply:
We need to have the calls to the cancel funcs here, otherwise these functions don't get called when the server exits due to ABORT.
The defer calls only run after the Wait() command in the Run() method, and this Wait never ends since the goroutines never complete (since cancel funcs don't get called).